fuse_core/src/transaction.cpp

```diff
@@ -127,6 +128,12 @@ Transaction::const_variable_range Transaction::addedVariables() const
     boost::make_transform_iterator(added_variables_.cend(), to_variable_ref));
 }
 
+bool Transaction::empty() const
+{
+  return boost::empty(added_variables_) && boost::empty(removed_variables_) && boost::empty(added_constraints_) &&
+         boost::empty(removed_constraints_);
```

Comment:
I believe the `involved_stamps_` should always be empty when the others are, but I don't mind adding `&& involved_stamps_.empty()`. Indeed, since `addInvolvedStamp` is used for that, it's probably better to check that.

Either way, it doesn't impact this PR.

Reply:
Added the `involved_stamps_`  check to the `empty()` method, and also added a unit test for `empty()`.